src/librustdoc/clean/render_macro_matchers.rs

```diff
@@ -63,7 +63,21 @@ fn snippet_equal_to_token(tcx: TyCtxt<'_>, matcher: &TokenTree) -> Option<String
 
     // Create a Parser.
     let psess = ParseSess::new(rustc_driver::DEFAULT_LOCALE_RESOURCES.to_vec());
-    let file_name = source_map.span_to_filename(span);
+
+    // Get the original filename and ensure we're using a fresh, unmapped filename
+    // to avoid trying to remap an already remapped filename
+    let file_name = match source_map.span_to_filename(span) {
+        FileName::Real(real_filename) => {
+            if let Some(local_path) = real_filename.clone().into_local_path() {
```

Comment:
`into_local_path()` can fail when "if the file was imported from another crate".

Which would be a issue with re-export of `macro_rules!`, wouldn't it?

Reply:
IMO using `FileName::macro_expansion_source_code(&snippet)` right away solves this, plus it is a safer approach.